### PR TITLE
Upgrade to zig 0.9.0

### PR DESCRIPTION
**Pull request by adyxax**
Hello,

Thanks you for writing this minesweeper! I stumbled on this repository while learning about zig and looking for examples. It compiles and works fine on 0.8.1 but not on the latest 0.9.0.

This patch fixes the compilation errors, essentially about https://ziglang.org/download/0.9.0/release-notes.html#Allocgate which changes the way allocators are passed to functions. The other changes are about the strictness of const and unused variables.

Warm regards,

### CODE REVIEW

File: src/help.zig [hunk 1]
```diff
@@ -48,7 +48,7 @@ const text_with_single_newlines =
     ;
 
 // does nothing to \n\n repeated, but replaces single \n with spaces
-fn removeSingleNewlines(s: []const u8, allocator: *std.mem.Allocator) ![]u8 {
+fn removeSingleNewlines(s: []const u8, allocator: std.mem.Allocator) ![]u8 {
```
Comment (Akuli) [inline src/help.zig:51]: Is passing the allocator by value surely a good idea? Is it now the recommended way?

**Review by Akuli — COMMENTED** [review]
Thanks! Please also change 0.8.1 in the README to 0.9.0.

File: src/help.zig (hunk 1 above)
Comment (Akuli) [reply src/help.zig:51]: It is, I just clicked your link

**Comment by adyxax** [comment]
> Thanks! Please also change 0.8.1 in the README to 0.9.0.

Good catch, I updated it.

**Comment by Akuli** [comment]
Thanks!

**Pull request merged**